### PR TITLE
[Tune] Pin Tune Dependencies

### PR DESCRIPTION
**Pull request by amogkam**
This PR pins all of the Tune dependencies. It is the first step in making sure that we have stable tests. Note that this currently does not pin transitive dependencies (subdepedencies) using pip-tools yet. First want to get a basic thing working with Dependabot and then will add that later.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
@ijrsvt do you know what's going on with this error?

```
Docker affected: True
Pulling images for caching
BUILT:  rayproject/base-deps:nightly-cpu
BUILT:  rayproject/base-deps:nightly-gpu
BUILT:  rayproject/ray-deps:nightly-cpu
BUILT:  rayproject/ray-deps:nightly-gpu
BUILT:  rayproject/ray:nightly-cpu
BUILT:  rayproject/ray:nightly-gpu
ERROR building: rayproject/ray-ml:nightly-cpu & error below:
{"stream":"Step 1/7 : ARG GPU"}
{"stream":"\n"}
{"stream":"Step 2/7 : FROM rayproject/ray:nightly\"$GPU\""}
{"stream":"\n"}
{"stream":" ---\u003e 9c0db37bbace\n"}
{"stream":"Step 3/7 : COPY requirements.txt ./"}
{"stream":"\n"}
{"stream":" ---\u003e 359a99049cae\n"}
{"stream":"Step 4/7 : COPY requirements_ml_docker.txt ./"}
{"stream":"\n"}
{"stream":" ---\u003e 54a318b43571\n"}
{"stream":"Step 5/7 : COPY requirements_rllib.txt ./"}
{"stream":"\n"}
{"stream":" ---\u003e e337f38736b4\n"}
{"stream":"Step 6/7 : COPY requirements/requirements_tune.txt ./"}
{"stream":"\n"}
{"errorDetail":{"message":"COPY failed: stat /var/lib/docker/tmp/docker-builder127233795/requirements/requirements_tune.txt: no such file or directory"},"error":"COPY failed: stat /var/lib/docker/tmp/docker-builder127233795/requirements/requirements_tune.txt: no such file or directory"}
```
I moved `requirements_tune.txt` into its own `requirements` directory and made the change in the corresponding Dockerfile. Is there anything else I have to change to get this working? Thanks.

**Comment by amogkam** [comment]
Tune, SGD, and Doc tests are all passing.

Other than the Docker image build it looks like the only other tests that are failing are `test_object_spilling`, `new_dashboard:test_job`, and Java tests. `test_object_spilling` is a known flaky test. Not sure about the latter 2, but they should be unrelated to these changes.

**Comment by ijrsvt** [comment]
@amogkam I think the fix in this [PR (against your branch) should fix it](https://github.com/amogkam/ray/pull/2/files). Basically we need to copy the file over first (in `build-docker-images.py`).

**Comment by amogkam** [comment]
Ah got it; thanks @ijrsvt!

**Comment by ijrsvt** [comment]
No problem @amogkam Just ping me if it still fails--it takes a try or two to get this working sometimes.

File: docker/ray-ml/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ARG GPU
 FROM rayproject/ray:nightly"$GPU"
 
+RUN mkdir ./requirements
 # We have to uninstall wrapt this way for Tensorflow compatibility
 COPY requirements.txt ./
 COPY requirements_ml_docker.txt ./
 COPY requirements_rllib.txt ./
-COPY requirements_tune.txt ./
+COPY requirements_tune.txt ./requirements/requirements_tune.txt
```
Comment (simon-mo) [inline docker/ray-ml/Dockerfile:9]: this is flipped? /requirements/requirements_tune should come from the host right?

File: docker/ray-ml/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
 ARG GPU
 FROM rayproject/ray:nightly"$GPU"
 
+RUN mkdir ./requirements
```
Comment (simon-mo) [inline docker/ray-ml/Dockerfile:4]: probably don't need this, a flat file is file

**Review by simon-mo — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]
looks fine to me!

**Pull request merged**